artemis-features/pom.xml

```diff
@@ -49,6 +49,58 @@
             <type>kar</type>
             <scope>provided</scope>
         </dependency>
+
+        <!-- these project-specific dependencies are referenced in features.xml and therefore should be declared here -->
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-quorum-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-server-osgi</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-core-client-osgi</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-jms-client-osgi</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-amqp-protocol</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-stomp-protocol</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-mqtt-protocol</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-hqclient-protocol</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-hornetq-protocol</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-openwire-protocol</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```

Comment:
I'm guessing you are adding this for build-ordering reasons? If so, it may be wise to structure things so that they apply build constraints without really altering the user-facing dependencies, e.g excluding all the deps (and perhaps making them optional also).

The fact these have never been here mean they clearly arent needed by the actual feature usage. Their new presence however could mean they are now all depended on for things touching the pom, and may need to resolve all those modules and their dependencies, even if the feature being used doesnt need them.